lib/dealership.rb

```diff
@@ -0,0 +1,68 @@
+class Dealership
+  attr_accessor :inventory, :name, :location, :lot_value
+
+  def initialize(name, location)
+    @name = name
+    @location = location
+    @inventory = []
+    @lot_value = 0
+  end
+
+  def inventory_count
+    @inventory.length
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    if inventory_count == 0
+      false
+    else
+      true
+    end
+  end
+
+  def cars_by_make(string)
+    cars_with_matching_make = []
+    @inventory.each do |car|
+      if car.make == string
+        cars_with_matching_make << car
+      end
+    end
+    return cars_with_matching_make
+  end
+
+  def total_value
+    @inventory.each do |car|
+      @lot_value += car.total_cost
+    end
+    return @lot_value
+  end
+
+  def details
+    details_hash = {}
+    details_hash['total_value']= @lot_value
+    details_hash['address']= @location
+    return details_hash
+  end
+
+  def average_price_of_car
+    @lot_value / inventory_count
```

Comment:
Again you will want to use your `total_value` method and divide that by `inventory_count`. But that returns an integer and we want the string "39,000". 

Look into the `.insert` method for strings 
https://ruby-doc.org/core-2.6/String.html#:~:text=to%20toggle%20source-,Inserts,-other_str%20before%20the